test/models/user_test.rb

```diff
@@ -4,8 +4,13 @@ class UserTest < ActiveSupport::TestCase
 
```

Comment:
I see your objective is to write thorough tests and I am all for it. We just need to make our tests more exclusive than what it is right now. There are some things that are not supposed to be tested here in the user model (as you will see in the comments). For guidance the two aspects we need to test for models generally are validations and associations. For validations it's simple (and you've done them). For associations I suggest you use the `shoulda` gem as it makes it simpler. As it is I am having a hard time understanding why some of the tests are in here.